R/ivreg.marg.R

```diff
@@ -0,0 +1,160 @@
+#' @export
+#' @rdname marg
+#' @examples
+#'
+#' # --------------------------------------
+#' # 2SLS using AER::ivreg (example adapted from ?AER::ivreg)
+#'
+#' data(margex)
+#' margex$assign <- margex$treatment
+#'
+#' # Probability of getting treatment increases with age and distance
+#' margex$pr_treat <- plogis(
+#'   margex$age / sd(margex$age) +
+#'   margex$distance / sd(margex$distance))
+#'
+#' # One-way non-compliance
+#' margex$actual <- margex$assign
+#' margex$actual[margex$assign == 1] <- rbinom(
+#'   n = sum(margex$assign == 1), size = 1,
+#'   prob = margex$pr_treat[margex$assign == 1]
+#' )
+#'
+#' mod <- AER::ivreg(y ~ as.factor(actual) + age + distance |
+#'                     as.factor(assign) + age + distance,
+#'                   data = margex)
+#'
+#'
+#' v <- function(object, ...){
+#'   vcov(object) * object$df.residual / nobs(object)
+#' }
+#'
+#' summary(mod, vcov = v)
+#'
+#' # Get the level of the outcome variable at different values of `actual`
+#' # Note data is a *required* parameter here
+#' marg(mod, var_interest = 'actual', data = margex, type = 'levels')
+#' marg(mod, var_interest = 'actual', data = margex, type = 'effects')
+#'
+marg.ivreg <- function(mod, var_interest,
+                       type = 'levels',
+                       vcov_mat = NULL,
+                       dof = NULL,
+                       at = NULL, base_rn = 1,
+                       at_var_interest = NULL,
+                       data,
```

Comment:
I tried to move it up the list for that reason, but the `cmd-check` yelled at me for violating S3 convention by having parameters in different orders... I wasn't sure what the best way to approach this is. 

I actually thought about setting the default to `NULL` but giving an error if `data` is `NULL` but that seemed... perverse.